core/init.c

```diff
@@ -36,6 +36,14 @@
 #include <auto_init.h>
 #endif
 
+#ifndef CONFIG_BOOT_MSG
+#define CONFIG_BOOT_MSG 1
+#endif
+
+#if !defined(CONFIG_BOOT_MSG_STRING) &&  CONFIG_BOOT_MSG
```

Comment:
Why check for `CONFIG_BOOT_MSG` here? If the macro is not used (i.e. when `CONFIG_BOOT_MSG=0` it should have no effect.

Reply:
fixed